File: ShoesStore/ShoesStore/ViewModel/RequestModel/ProductViewModel.cs
using ShoesStore.Model;

namespace ShoesStore.ViewModel.RequestModel
{
    public class ProductViewModel
    {
		public int Id { get; set; }
		public string Name { get; set; }
		public string Description { get; set; }
		public decimal Price { get; set; }
		public int CategoryId { get; set; }
		public string CategoryName { get; set; }
		public DateTime CreatedAt { get; set; }
		public DateTime UpdatedAt { get; set; }

		public List<int> SizeId { get; set; } 
		public List<string> SizeName { get; set; } 
		public List<int> Quantity { get; set; }
		public string Url { get; set; }
	}
}

File: UI/src/app/users/models/product.viewmodel.ts
export interface ProductViewModel {
  id: number;
  name: string;
  description: string;
  price: number;
  categoryId: number;
  categoryName: string;
  createdAt: string; // ISO string format
  updatedAt: string; // ISO string format
  sizeId: number[];
  sizeName: string[];
  quantity: number[];
  url: string; // Có thể null nếu không có hình ảnh
}

export interface PaginatedResponse<T> {
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
  data: T[];
}
